modules/porous_flow/src/actions/PorousFlowActionBase.C

```diff
@@ -99,9 +101,13 @@ PorousFlowActionBase::PorousFlowActionBase(const InputParameters & params)
     _num_aqueous_equilibrium(getParam<unsigned int>("number_aqueous_equilibrium")),
     _num_aqueous_kinetic(getParam<unsigned int>("number_aqueous_kinetic")),
     _gravity(getParam<RealVectorValue>("gravity")),
-    _mass_fraction_vars(getParam<std::vector<VariableName>>("mass_fraction_vars")),
+    _mass_fraction_vars(isParamValid("mass_fraction_vars")
+                            ? getParam<std::vector<VariableName>>("mass_fraction_vars")
+                            : std::vector<VariableName>{}),
```

Comment:
use default in parameter

Reply:
Default can not be used here as well. See line 395, they want to error out when no value is set for "mass_fraction_vars". I don't think I should check that

Reply:
yeah it s not a clear winner here between your code and changing the check at line 395. So let s keep that